### ISSUE TITLE
Not all records imported into View by Property 

### ISSUE THREAD

**Issue by RDmitchell** [SEED-platform/seed#1089]
Not all the records are imported into the Property tab from the example property spreadsheet

I am testing on seeddemostaging on 10/10/2016 with Nathan's latest matching code up (I think). I made a completely new organization, and this is the first and only file imported into it.

The screenshot below shows the records in the original spreadsheet and the records that are showing up in the View by Property tab after importing, mapping and matching (matching didn't seem to work).

![image](https://cloud.githubusercontent.com/assets/6314950/19249718/c378d806-8eeb-11e6-94c9-3af695d614f4.png)

Here is how I mapped the data.  Not sure what the program will do with the Extra Data fields because it is not possible to assign them to a Tax Lot or Property table, but for now I will not worry about that, as it doesn't seem to affect the outcome. 
![image](https://cloud.githubusercontent.com/assets/6314950/19250426/0e8c68f0-8eef-11e6-935e-45b2c480e95d.png)

Click the Map your Data button -- shows the 2 extra data fields, and indicates that there are 12 records, which is correct
![image](https://cloud.githubusercontent.com/assets/6314950/19250441/3033e1cc-8eef-11e6-8710-8296d189b2fb.png)

Click the Save Mappings and Confirm matching buttons
Go to Inventory View by Property, the program reports 10 records instead of 12 -- comparing to the spreadsheet, the missing PM Property IDs are
* 1311525
* 1311528
![image](https://cloud.githubusercontent.com/assets/6314950/19250542/b4566164-8eef-11e6-9ede-ad8112e5dcc0.png)

Here is a screen shot of the original spreadsheet
![image](https://cloud.githubusercontent.com/assets/6314950/19250595/0459467c-8ef0-11e6-982f-2424efad2757.png)

**Comment by RDmitchell** [comment]
It might be worth figuring out the issue here before the demo on Oct 13th if possible. I can't really show the original spreadsheet and the results of the import if this bug is not fixed. I can work around it by not showing the import, mapping and matching functionality for the demo -- we will just have to see how far @maronnax and @nllong get with figuring out this issue.

**Comment by RDmitchell** [comment]
Here is a link to the example file
https://drive.google.com/open?id=0B3fTKpZ9Dx7LUHpOaTFLRWMtVW8

I can also make anyone a member of this test organization if you want to view the data in the org itself.

**Comment by RDmitchell** [comment]
@maronnax pointed out that the problem is that the records that are not showing up both have the same ID number, so they are being merged. So I am closing this and will test with a proper set of example data.

**Comment by RDmitchell** [comment]
The sample Property data has duplicate Custom IDs for a few of the records. This might be interesting to add to a test case, but I don't think this data set represents that properly. 

So I am going to try 2 tests:
* Don't map custom ID as a matching field -- this should allow the program to bring in all 12 records and not do any matching on that custom ID field. At least that is how the program should work

* Delete the custom ID column altogether, and just verify that the program will import all 12 records if the Custom ID is not there.

Then based on these test results, we need to figure out a good test case where there is a Custom ID is in a file in a manner that a user would do it. I am not sure how that would now be used, given that at this point we have a field for unique building id, I think, ie, "jurisdiction_property_id" -- right ? That is how Custom ID used to be used, but that is no longer the case hopefully.

**Comment by RDmitchell** [comment]
Without the Custom ID, the program is still matching on the Address field, and two of the records have the same address, so it is combining them into one record. So that makes sense from a merging standpoint, but it seems like that should be reflected in the Detail view, ie, both records should appear as columns with one of them being the "current" version -- or the merged version being the "current" version (not sure how that works).

Here is the data file in Excel -- you can see that line 8 and 9 both have the same address.
These two PM records are different buildings but have the same address. 

* I believe the program is matching on address and then creating one record from the two records. 
* However, this is probably not the desired behavior, since they are two different buildings.
* At the moment probably the only work around to prevent this matching is to not map the address to a matching field, which is what users do in V 1.5 
![image](https://cloud.githubusercontent.com/assets/6314950/19289764/2540da86-8fc1-11e6-8846-235507aa0f38.png)

When this file is imported and displayed in the Inventory View by Property tab, the record with PM Property ID 1311528 is missing.
![image](https://cloud.githubusercontent.com/assets/6314950/19289882/b0c6752a-8fc1-11e6-8430-3def61b88949.png)

So presumably the program has "merged" PM Property ID 1311527 and 1311528 into one record. 

In the Detail view only data for 1311527 is being displayed.

* If it merged the record with PM Property ID 1311528 into this record, there should be a column to the right of Current Value showing the data from 1311528. That is how V 1.5 works.
* Otherwise how does the user know what happened to 1311528?

![image](https://cloud.githubusercontent.com/assets/6314950/19290009/4d692a9e-8fc2-11e6-9793-f82314ed6b42.png)

**Comment by RDmitchell** [comment]
A few more tests:

* don't make Address a matching field and see what the results are
* see how V 1.5 would treat this case

**Comment by RDmitchell** [comment]
First Test -- don't map by address field 
![image](https://cloud.githubusercontent.com/assets/6314950/19290508/e21b553e-8fc4-11e6-8ad7-597aa70da898.png)
In the Matching review screen, where only the fields in Extra Data are displayed, Address shows up, so it has been added to the Extra Data field and presumably the program won't match on it. 
![image](https://cloud.githubusercontent.com/assets/6314950/19290601/4e63ad0e-8fc5-11e6-9374-9654613bccb8.png)
Looking at the Inventory View by Property tab, now all 11 records are there. 

However, there is a problem with what field is showing the address from the data. It should be the newly mapped "Address" field, but it seems that it is actually "address_line_1" which is incorrect.  I will make a separate issue for this.
![image](https://cloud.githubusercontent.com/assets/6314950/19291027/9af23efe-8fc7-11e6-9985-fb9e4722b272.png)

**Comment by RDmitchell** [comment]
The problem with what field is displaying the address data is in a new issue
https://github.com/SEED-platform/seed/issues/1094

**Comment by rebrownlbnl** [comment]
So the logic should be that if two records have different values for pm_property_id, they should not be merged into the same property view, even if the address is the same? So pm_property_id is kind of a "master" field for determining if two property_state records should be merged?

**Comment by RDmitchell** [comment]
I'm not sure we know what the logic is.

What we have now is the same basic situation as in V 1.5, where you have to pick and choose which matching fields to use. 

The REAL solution is to allow the users to match any field to any field between data files, and have an order of matching that the user has control over. We have talked about this for a long time (including in sessions with users where they have given useful feedback), and I think it is something that we need to work out and implement for FY 2017.

**Comment by nllong** [comment]
it appears the issue is related to these:
![image](https://cloud.githubusercontent.com/assets/1907354/19326261/0542b530-9087-11e6-8b68-0da0e44d9ca3.png)
I'm digging into it.

**Comment by RDmitchell** [comment]
The reason that 1311528 is missing is because the addresses are the same and mapped to Address_line_1, so I believe Nathan's code merges them into one record. 

When I change the addresses to be different, then all the records appear. 

Not sure we need to dig into this right now -- not critical for the demo. For the sample data, I just made the addresses different.

The real issue with this bug is why the data ends up under Address_line_1 instead of Address in the list display. Because I mapped it to "Address" in order to not have the program merge the records.
